### ISSUE TITLE
read variable?"Prompt" format errors with SC2154 in ksh+ksh93

### ISSUE THREAD

**Issue by d-little** [koalaman/shellcheck#2272]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2154
- My shellcheck version (`shellcheck --version` or "online"):  online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

These two are functionally identical in ksh/ksh93 and bash:

```sh
#!/usr/bin/ksh93
read -r example?"Text before user input: "
if (( "${#example}" == 0 )); then 
	printf "%s\n" "No user input"
else
	printf "%s\n" "${example}"
fi
```

```sh
#!/usr/bin/bash
printf "%s: " "Text before user input: "
read -r example
if (( "${#example}" == 0 )); then 
	printf "%s\n" "No user input"
else
	printf "%s\n" "${example}"
fi
```

#### Here's what shellcheck currently says:

Marks the line `if (( "${#example}" == 0 )); then` as SC2154, but this is not true in KSH93

#### Here's what I wanted or expected to see:

No error.  The read variable?"Prompt" format is a valid method in ksh/ksh93
